### ISSUE TITLE
Can't detect memory location when overriding emulated memory size

### ISSUE THREAD

**Issue by NerduMiner** [bkacjios/m-overlay#28]
M'Overlay works just fine on Project Slippi on my machine on both Manjaro Linux and Windows 10. However, if I try to run the program with the latest development version of Dolphin 5.0(Running the same iso of Melee) the controller overlay won't appear. The most that will happen is the console showing that the program has hooked onto a dolphin process:
```
[INFO  - 15:47:07] [DOLPHIN] Hooked
```
I'm not sure if I am doing anything wrong, I was planning to add support for Pikmin 2 for when I partake in a race of a cave mod which is how I discovered this issue.

**Comment by bkacjios** [comment]
That's strange. It seems to be working fine for me on the latest dolphin development build, specifically Dolphin 5.0-12367

```[INFO  - 16:06:20] [DOLPHIN] Hooked
[INFO  - 16:06:23] [DOLPHIN] Watching ram: 9E910000
[DEBUG - 16:06:24] [DOLPHIN] GAMEID: "GALE01" (Version 2)
[INFO  - 16:06:24] [DOLPHIN] Loaded game config: GALE01-2
[INFO  - 16:06:24] [MEMORY] Mapping game memory..
[DEBUG - 16:06:24] [MEMORY] [0][8046B6AE = 00000000] match.finished = false
[DEBUG - 16:06:24] [MEMORY] [0][80479D33 = 00000000] menu = 0
[DEBUG - 16:06:24] [MEMORY] [0][8046B6A5 = 00000000] match.started = false```

**Comment by bkacjios** [comment]
I have Dolphin 5.0-12071 on my Arch Linux install, and it seems to be hooking it just fine for me.

![Linux Dolphin Hooking/Running](https://i.imgur.com/DOgZRPy.png)

**Comment by NerduMiner** [comment]
It's happening on both Windows and Linux for me. This is what I get when I run on Windows 10 currently with the latest dev version. I used the installer to install m'overlay
![image](https://user-images.githubusercontent.com/23490112/89109886-33fe6a80-d413-11ea-858c-60d208848b55.png)

**Comment by bkacjios** [comment]
You're not running dolphin as admin or anything are you?

**Comment by NerduMiner** [comment]
Nope, it wouldn't hook otherwise

**Comment by bkacjios** [comment]
Hmm.. That's really strange. The only other thing I could think of you have a second Dolphin.exe running and it's hooking that instead of the current one you're using.

**Comment by NerduMiner** [comment]
I've made sure that I'm only running one at a time, I have no clue why it's not working for me. I'll try running this on a different machine that has Windows 10 and Linux Mint and if it works on both of those then its just something weird with my desktop.

**Comment by NerduMiner** [comment]
On my laptop, the program adds a "watching ram:" log on Windows 10, but more or less acts the same as my desktop. I couldn't get the latest version of love2d to install on Linux Mint so I couldn't test. It seems I am having problems with both machines, I don't really know what's causing it so if it's fine with you I can close this issue.

**Comment by NerduMiner** [comment]
Actually, i figured out what's wrong. In newer versions of Dolphin there is the option to increase the amount of memory the emulated gamecube can use.
![image](https://user-images.githubusercontent.com/23490112/89110932-1b478200-d41e-11ea-800d-5f2dbfe7860d.png)
I had this enabled on both OS's because this setting is necessary for getting the Pikmin 2 cave mod to run.

**Comment by bkacjios** [comment]
Oh, that makes sense actually. Since you're changing the size, it can't detect the memory space properly. Basically, M'Overlay is programmed to find a memory space in dolphin that is equal to the gamecube's max ram size.

https://github.com/bkacjios/m-overlay/blob/master/source/modules/memory/windows.lua#L264

This value is currently hard coded, but I could maybe add a commandline option to have it look for different ram sizes for cases like this.

**Comment by bkacjios** [comment]
After experimenting with that setting in dolphin, I may be able to get away with changing the check to a greater than or equal to. I need to test it some more to make sure it's not hooking some other section of dolphins memory, but it seems to be working.

**Comment by bkacjios** [comment]
In 1.4.0 it will be able to hook into the game no matter what custom emulated memory size you have set.

**Comment by NerduMiner** [comment]
Thanks for adding support for the Pikmin games! I was able to figure out Pikmin 2 beforehand but I got stuck because some mods run on a demo version of the game which has the same revision and game id as the release version but stores controller inputs in a different location. I wanted to have support for both versions with just one file but I couldn't figure out how to read the gamecube's ram for a version check.
The demo version stores inputs at 0x80506F48 and a memory check at 0x8005426 could determine whether the current version is the demo(C5) or the release version(C6)
